src/views/CreateTransformer.vue

```diff
@@ -0,0 +1,175 @@
+    <ListSection
+      :title="`Attributes of ${newTransformer.resourceType.name}`"
+      :list="resourceTypeAttributeList"
+    />
+    <br>
+    <h1>below you can enter the code of your transformer</h1>
+    <br>
+    <TextArea
+      :firstString="`instancesOf${newTransformer.resourceType.name}.${newTransformer.transformerType}( (instance) => {`"
+      secondString='});'
+      :value.sync="newTransformer.body"
+      placeholder="return (instances.getAttribute(age) > 18);"
+    />
+    <Button text="Create Transformer" :onClick="createTransformer"/>
+  </main>
+
+</template>
+
+<script lang="ts">
+import { Component, Vue } from 'vue-property-decorator';
+import {
+  ResourceTypes,
+  ResourceType,
+  Transformer,
+  Transformers,
+  createTransformerNullObject,
+  ResourceTypeNullObject,
+} from '@/apis/rembrandt/rembrandt';
+import Select, { Option } from '@/components/Select.vue';
+import Li, { ListEntry } from '@/components/Li.vue';
+import ViewHeader from '@/components/ViewHeader.vue';
+import ListSection from '@/components/ListSection.vue';
+import Input from '@/components/Input.vue';
+import Button from '@/components/Button.vue';
+import Utils from '@/utils/Utils';
+import { NotificationLevel } from '@/interfaces/Notification';
+import { transformerTypes } from '@/apis/rembrandt/rembrandt';
+import TextArea from '@/components/TextArea.vue';
+
+@Component({
+  components: {
+    ListSection,
+    Li,
+    ViewHeader,
+    Input,
+    Button,
+    Select,
+    TextArea,
+  },
+})
+export default class CreateTransformer extends Vue {
+  // region public static methods
+  // endregion
+
+  // region private static methods
+  // endregion
+
+  // region public members
+  public newTransformer: Transformer;
+  public formState: number = 0;
+  public resourceType: ResourceType;
+  public resourceTypes: ResourceType[] = [];
+
+  public get resourceTypesList(): ListEntry[] {
+    return Utils.resourceTypesToList(this.resourceTypes, this.selectResourceType);
+  }
+
+  public get resourceTypeAttributeList(): ListEntry[] {
+    if (!this.newTransformer.resourceType) {
+      return [];
+    }
+    return Utils.resourceTypeAttributesToList(this.newTransformer.resourceType);
+  }
+
+  // endregion
+
+  // region private members
+  // endregion
+
+  // region constructor
+  constructor() {
+    super();
+    this.newTransformer = createTransformerNullObject();
+    this.resourceType = ResourceTypeNullObject;
+  }
+  // endregion
+
+  // region public methods
+  public async mounted() {
+    this.resourceTypes = await ResourceTypes.get();
+    this.newTransformer.transformerType = 'map';
+  }
+
+  public previousStep(): void {
+    if (this.formState > 0) {
+      this.formState--;
+    }
+  }
+
+  public nextStep(): void {
+    this.formState++;
+  }
+
+  public async selectResourceType(id: string) {
+    this.resourceType = await ResourceTypes.getOne(id);
```

Comment:
What for do we need `this.resourceType`?
If I haven't overlooked anything, this is not used again. (At least it is a duplicate with `this.newTransformer.resourceType`.)

Reply:
it was used to later assign the resourceType to the new transformer, but since it can be assigned directly, I removed it.